src/vs/workbench/contrib/accessibility/browser/accessibleView.ts

```diff
@@ -0,0 +1,160 @@
+	_serviceBrand: undefined;
+
+	private _providers: Map<string, IAccessibleContentProvider> = new Map();
+	private _editorWidget: CodeEditorWidget;
+	private _editorContainer: HTMLElement;
+
+	constructor(
+		@IInstantiationService private readonly _instantiationService: IInstantiationService,
+		@IConfigurationService private readonly _configurationService: IConfigurationService,
+		@IModelService private readonly _modelService: IModelService,
+		@IContextViewService private readonly _contextViewService: IContextViewService
+	) {
+		super();
+		this._editorContainer = document.createElement('div');
+		this._editorContainer.classList.add('accessible-view');
+		const codeEditorWidgetOptions: ICodeEditorWidgetOptions = {
+			contributions: EditorExtensionsRegistry.getSomeEditorContributions([LinkDetector.ID, SelectionClipboardContributionID, 'editor.contrib.selectionAnchorController'])
+		};
+		const editorOptions: IEditorConstructionOptions = {
+			...getSimpleEditorOptions(),
+			lineDecorationsWidth: 6,
+			dragAndDrop: true,
+			cursorWidth: 1,
+			wrappingStrategy: 'advanced',
+			wrappingIndent: 'none',
+			padding: { top: 2, bottom: 2 },
+			quickSuggestions: false,
+			renderWhitespace: 'none',
+			dropIntoEditor: { enabled: true },
+			accessibilitySupport: this._configurationService.getValue<'auto' | 'off' | 'on'>('editor.accessibilitySupport'),
+			cursorBlinking: this._configurationService.getValue('terminal.integrated.cursorBlinking'),
+			readOnly: true
+		};
+		this._editorWidget = this._register(this._instantiationService.createInstance(CodeEditorWidget, this._editorContainer, editorOptions, codeEditorWidgetOptions));
+	}
+
+	registerProvider(provider: IAccessibleContentProvider): void {
+		this._providers.set(provider.id, provider);
+	}
+
+	show(providerId: string): void {
+		const delegate: IContextViewDelegate = {
+			getAnchor: () => this._editorContainer,
+			render: (container) => {
+				return this._render(providerId, container);
+			},
+			onHide: () => {
+				this._providers.get(providerId)?.onClose();
+			}
+		};
+		this._contextViewService.showContextView(delegate);
+	}
+
+	private _getProviderOrThrow(providerId: string): IAccessibleContentProvider {
+		const provider = this._providers.get(providerId);
+		if (!provider) {
+			throw new Error(`No accessible view provider with id: ${providerId}`);
+		}
+		return provider;
+	}
+
+	private _render(providerId: string, container: HTMLElement): IDisposable {
+
+		const provider = this._getProviderOrThrow(providerId);
+
+		let fragment = '';
+		if (provider.options?.isHelpMenu) {
+			fragment += localize('introMsg', "Welcome to {0} Accessibility Help. Exit this menu and return to the {0} via the Escape key.\n", providerId);
+		}
+		fragment += provider.provideContent();
+
+		this._getTextModel(URI.from({ path: `accessible-view-${providerId}`, scheme: 'accessible-view', fragment })).then((model) => {
```

Comment:
providerId could contain spaces, is that an issue?

Reply:
tested and it doesn't seem to be an issue